File: sdkwork-backend-sdk-java/src/main/java/com/sdkwork/backend/api/ChannelAccountApi.java
package com.sdkwork.backend.api;

import com.sdkwork.backend.http.HttpClient;
import com.sdkwork.backend.model.*;
import java.util.List;
import java.util.Map;

public class ChannelAccountApi {
    private final HttpClient client;
    
    public ChannelAccountApi(HttpClient client) {
        this.client = client;
    }

    /** Update channel account */
    public PlusApiResultPlusChannelAccountVO update(PlusChannelAccountForm body) throws Exception {
        return (PlusApiResultPlusChannelAccountVO) client.put(ApiPaths.backendPath("/channel/account"), body);
    }

    /** Create channel account */
    public PlusApiResultPlusChannelAccountVO create(PlusChannelAccountForm body) throws Exception {
        return (PlusApiResultPlusChannelAccountVO) client.post(ApiPaths.backendPath("/channel/account"), body);
    }

    /** Get SMS channel accounts */
    public PlusApiResultListPlusChannelAccountVO listSms(PlusChannelAccountQueryForm body) throws Exception {
        return (PlusApiResultListPlusChannelAccountVO) client.post(ApiPaths.backendPath("/channel/account/list_sms"), body);
    }

    /** Get payment channel accounts */
    public PlusApiResultListPlusChannelAccountVO listPayment(PlusChannelAccountQueryForm body) throws Exception {
        return (PlusApiResultListPlusChannelAccountVO) client.post(ApiPaths.backendPath("/channel/account/list_payment"), body);
    }

    /** Get OSS channel accounts */
    public PlusApiResultListPlusChannelAccountVO listOss(PlusChannelAccountQueryForm body) throws Exception {
        return (PlusApiResultListPlusChannelAccountVO) client.post(ApiPaths.backendPath("/channel/account/list_oss"), body);
    }

    /** Get official account channel accounts */
    public PlusApiResultListPlusChannelAccountVO listOfficialAccount(PlusChannelAccountQueryForm body) throws Exception {
        return (PlusApiResultListPlusChannelAccountVO) client.post(ApiPaths.backendPath("/channel/account/list_official_account"), body);
    }

    /** Get mini program channel accounts */
    public PlusApiResultListPlusChannelAccountVO listMiniProgram(PlusChannelAccountQueryForm body) throws Exception {
        return (PlusApiResultListPlusChannelAccountVO) client.post(ApiPaths.backendPath("/channel/account/list_mini_program"), body);
    }

    /** Get channel accounts by page */
    public PlusApiResultPagePlusChannelAccountVO listByPage(QueryListForm body, Map<String, Object> params) throws Exception {
        return (PlusApiResultPagePlusChannelAccountVO) client.post(ApiPaths.backendPath("/channel/account/list"), body, params);
    }

    /** Get all channel accounts */
    public PlusApiResultListPlusChannelAccountVO listAllEntities(QueryListForm body) throws Exception {
        return (PlusApiResultListPlusChannelAccountVO) client.post(ApiPaths.backendPath("/channel/account/list/all"), body);
    }

    /** Get channel account by ID */
    public PlusApiResultPlusChannelAccountVO getById(String id) throws Exception {
        return (PlusApiResultPlusChannelAccountVO) client.get(ApiPaths.backendPath("/channel/account/" + id + ""));
    }

    /** Delete channel account */
    public PlusApiResultBoolean delete(String id) throws Exception {
        return (PlusApiResultBoolean) client.delete(ApiPaths.backendPath("/channel/account/" + id + ""));
    }
}

File: sdkwork-backend-sdk-kotlin/src/main/kotlin/com/sdkwork/backend/api/ChannelAccountApi.kt
package com.sdkwork.backend.api

import com.sdkwork.backend.*
import com.sdkwork.backend.http.HttpClient

class ChannelAccountApi(private val client: HttpClient) {

    /** Update channel account */
    suspend fun update(body: PlusChannelAccountForm): PlusApiResultPlusChannelAccountVO? {
        return client.put(ApiPaths.backendPath("/channel/account"), body) as? PlusApiResultPlusChannelAccountVO
    }

    /** Create channel account */
    suspend fun create(body: PlusChannelAccountForm): PlusApiResultPlusChannelAccountVO? {
        return client.post(ApiPaths.backendPath("/channel/account"), body) as? PlusApiResultPlusChannelAccountVO
    }

    /** Get SMS channel accounts */
    suspend fun listSms(body: PlusChannelAccountQueryForm? = null): PlusApiResultListPlusChannelAccountVO? {
        return client.post(ApiPaths.backendPath("/channel/account/list_sms"), body) as? PlusApiResultListPlusChannelAccountVO
    }

    /** Get payment channel accounts */
    suspend fun listPayment(body: PlusChannelAccountQueryForm? = null): PlusApiResultListPlusChannelAccountVO? {
        return client.post(ApiPaths.backendPath("/channel/account/list_payment"), body) as? PlusApiResultListPlusChannelAccountVO
    }

    /** Get OSS channel accounts */
    suspend fun listOss(body: PlusChannelAccountQueryForm? = null): PlusApiResultListPlusChannelAccountVO? {
        return client.post(ApiPaths.backendPath("/channel/account/list_oss"), body) as? PlusApiResultListPlusChannelAccountVO
    }

    /** Get official account channel accounts */
    suspend fun listOfficialAccount(body: PlusChannelAccountQueryForm? = null): PlusApiResultListPlusChannelAccountVO? {
        return client.post(ApiPaths.backendPath("/channel/account/list_official_account"), body) as? PlusApiResultListPlusChannelAccountVO
    }

    /** Get mini program channel accounts */
    suspend fun listMiniProgram(body: PlusChannelAccountQueryForm? = null): PlusApiResultListPlusChannelAccountVO? {
        return client.post(ApiPaths.backendPath("/channel/account/list_mini_program"), body) as? PlusApiResultListPlusChannelAccountVO
    }

    /** Get channel accounts by page */
    suspend fun listByPage(body: QueryListForm? = null, params: Map<String, Any>? = null): PlusApiResultPagePlusChannelAccountVO? {
        return client.post(ApiPaths.backendPath("/channel/account/list"), body, params) as? PlusApiResultPagePlusChannelAccountVO
    }

    /** Get all channel accounts */
    suspend fun listAllEntities(body: QueryListForm? = null): PlusApiResultListPlusChannelAccountVO? {
        return client.post(ApiPaths.backendPath("/channel/account/list/all"), body) as? PlusApiResultListPlusChannelAccountVO
    }

    /** Get channel account by ID */
    suspend fun getById(id: String): PlusApiResultPlusChannelAccountVO? {
        return client.get(ApiPaths.backendPath("/channel/account/$id")) as? PlusApiResultPlusChannelAccountVO
    }

    /** Delete channel account */
    suspend fun delete(id: String): PlusApiResultBoolean? {
        return client.delete(ApiPaths.backendPath("/channel/account/$id")) as? PlusApiResultBoolean
    }
}
